zio/anyio/reader.go

```diff
@@ -32,6 +34,36 @@ func NewReaderWithOpts(zctx *zed.Context, r io.Reader, opts ReaderOpts) (zio.Rea
 	if opts.Format != "" && opts.Format != "auto" {
 		return lookupReader(zctx, r, opts)
 	}
+
+	var parquetErr, vngErr error
+	if rs, ok := r.(io.ReadSeeker); ok {
+		if n, err := rs.Seek(0, io.SeekCurrent); err == nil {
+			var zr zio.Reader
+			zr, parquetErr = parquetio.NewReader(zctx, rs)
+			if parquetErr == nil {
+				return zio.NopReadCloser(zr), nil
+			}
+			if _, err := rs.Seek(n, io.SeekStart); err != nil {
+				return nil, err
+			}
+			zr, vngErr = vngio.NewReader(zctx, rs)
+			if vngErr == nil {
+				return zio.NopReadCloser(zr), nil
+			}
+			if _, err := rs.Seek(n, io.SeekStart); err != nil {
+				return nil, err
+			}
+		} else {
+			parquetErr = err
+			vngErr = err
+		}
+		parquetErr = fmt.Errorf("parquet: %w", parquetErr)
+		vngErr = fmt.Errorf("vng: %w", vngErr)
+	} else {
+		parquetErr = errors.New("parquet: reader cannot seek")
```

Comment:
"reader cannot seek" is describing the implementation not the supported types of inputs begging the question "what's a reader?".  How about `seekable input required for auto-detection (consider "-i parquet")`

Reply:
@mccanne: Good point, though "consider '-i parquet'" isn't quite the right advice since that won't make the input seekable, and that's the requirement here. I'll go with "seekable input required for auto-detection" unless you've got other ideas.

Reply:
Done in f82fb695cb1b26ba86ccfab3abea21ece1376fbd, though the exact language I used is "auto-detection requires seekable input" to align with `lineErr := errors.New("line: auto-detection not supported")`.